src/PlaneWaveBasis.jl

```diff
@@ -143,63 +143,16 @@ build_kpoints(basis::PlaneWaveBasis, kcoords) =
                                 fft_size=nothing, variational=true,
                                 optimize_fft_size=false, supersampling=2,
                                 kgrid=nothing, kshift=nothing) where {T <: Real}
-    # TODO this constructor is too complicated, we should simplify it
-    # if possible (esp. the variational part)
     mpi_ensure_initialized()
-    if variational
-        @assert Ecut > 0
-        if fft_size === nothing
-            fft_size = compute_fft_size(model, Ecut; supersampling=supersampling)
-        end
-
-        if optimize_fft_size
-            # TODO This is a hack for now, we build the kpoints twice
-            fft_size = Tuple{Int, Int, Int}(fft_size)
-            kpoints = build_kpoints(model, fft_size, kcoords, Ecut; variational=variational)
-            fft_size = compute_fft_size_precise(model.lattice, Ecut, kpoints;
-                                                supersampling=supersampling)
-        end
-
-        # Sanity checks
-        max_E = sum(abs2, model.recip_lattice * floor.(Int, Vec3(fft_size) ./ 2)) / 2
-        Ecut > max_E && @warn(
-            "For a variational method, Ecut should be less than the maximal kinetic " *
-            "energy the grid supports ($max_E)"
-        )
-    else
-        # ensure fft_size is provided, and other options are not set
-        # TODO make proper error messages when the interface gets a bit cleaned up
-        @assert fft_size !== nothing
-        @assert supersampling == 2
-        @assert !optimize_fft_size
-    end
-
-    # TODO generic FFT is kind of broken for some fft sizes
-    #      ... temporary workaround, see more details in fft_generic.jl
-    fft_size = next_working_fft_size.(T, fft_size)
-    fft_size = Tuple{Int, Int, Int}(fft_size)
-    (ipFFT_unnormalized,  opFFT_unnormalized,
-     ipBFFT_unnormalized, opBFFT_unnormalized) = build_fft_plans(T, fft_size)
-    # The FFT interface specifies that fft has no normalization, and
-    # ifft has a normalization factor of 1/length (so that both
-    # operations are inverse to each other). The convention we want is
-    # ψ(r) = sum_G c_G e^iGr / sqrt(Ω)
-    # so that the ifft is normalized by 1/sqrt(Ω). It follows that the
-    # fft must be normalized by sqrt(Ω) / length
-    ipFFT = ipFFT_unnormalized * (sqrt(model.unit_cell_volume) / length(ipFFT_unnormalized))
-    opFFT = opFFT_unnormalized * (sqrt(model.unit_cell_volume) / length(opFFT_unnormalized))
-    ipIFFT = inv(ipFFT)
-    opIFFT = inv(opFFT)
 
-    # Compute kpoint information
+    #### Compute kpoint information
```

Comment:
I'm not a fan of this multiple `####` thing. It's only opening the room for all sorts of messes (Is it 3 or 4 `#`, how about 2 is that a more important comment than `#`. I think with the changes you made there is no real need for this. And either a function is short enough that the structure is clear or you ought to split it up.

Reply:
I think I agree. I put them before the refactor but now there's less need